custom-templates/priv_instances_public_nlb_ssl/neo4j-nlb-priv-ssl.template.yaml

```diff
@@ -0,0 +1,857 @@
+  Neo4jInstanceSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Security group for instances, open across the VPC
+      VpcId: !Ref Neo4jVPC
+      SecurityGroupIngress:
+        - IpProtocol: -1
+          CidrIp: !GetAtt Neo4jVPC.CidrBlock
+        - IpProtocol: tcp
+          FromPort: 7474
+          ToPort: 7474
+          CidrIp: 0.0.0.0/0
+        - IpProtocol: tcp
+          FromPort: 7687
+          ToPort: 7687
+          CidrIp: 0.0.0.0/0
+
+  #Bastion Security Group
+  Neo4jBastionSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Security Group for Bastion Instance
+      VpcId: !Ref Neo4jVPC
+      SecurityGroupIngress:
+      - IpProtocol: tcp
+        FromPort: 22
+        ToPort: 22
+        CidrIp: !Ref BastionSSHCIDR
+
+  Neo4jAutoScalingGroup:
+    Type: AWS::AutoScaling::AutoScalingGroup
+    DependsOn: Neo4jNATGateway
+    CreationPolicy:
+      ResourceSignal:
+        Count: !Ref NumberOfServers
+        Timeout: PT8M
+    Properties:
+      AvailabilityZones: !If [CreateCluster, [!Select [ 0, Fn::GetAZs: !Ref 'AWS::Region' ] , !Select [ 1, Fn::GetAZs: !Ref 'AWS::Region' ] , !Select [ 2, Fn::GetAZs: !Ref 'AWS::Region' ]],[!Select [ 0, Fn::GetAZs: !Ref 'AWS::Region' ]]]
+      LaunchConfigurationName:
+        Ref: Neo4jLaunchConfiguration
+      MinSize: !Ref NumberOfServers
+      MaxSize: !Ref NumberOfServers
+      VPCZoneIdentifier: !If [CreateCluster, [!Ref Neo4jPrivateSubnet1,!Ref Neo4jPrivateSubnet2,!Ref Neo4jPrivateSubnet3], [!Ref Neo4jPrivateSubnet1]]
+      TargetGroupARNs:
+        - Ref: Neo4jHTTPTargetGroup
+        - Ref: Neo4jBoltTargetGroup
+      DesiredCapacity:
+        !Ref NumberOfServers
+      Tags:
+        - Key: StackID
+          Value: !Ref 'AWS::StackId'
+          PropagateAtLaunch: true
+        - Key: Name
+          Value: !Ref 'AWS::StackName'
+          PropagateAtLaunch: true
+
+  Neo4jInstanceProfile:
+    Type: AWS::IAM::InstanceProfile
+    Properties:
+      Roles:
+        - Ref: Neo4jRole
+  Neo4jRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - ec2.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Policies:
+        - PolicyName: Neo4jPolicy
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - autoscaling:DescribeAutoScalingInstances
+                  - cloudformation:DescribeStack*
+                  - ec2:DescribeInstances
+                Resource: "*"
+
+ #Bastion (Public) EC2 Instance 
+  Neo4jBastionInstance: 
+    Type: AWS::EC2::Instance
+    Properties: 
+      SubnetId: !Ref Neo4jPublicSubnet1
+      InstanceType: t3.micro
+      ImageId: "ami-09d3b3274b6c5d4aa"
```

Comment:
I guess this needs to be parameterised. Also, I tried running `priv_instance_public_nlb/neo4j-new.template.yaml` and it failed to create the Bastion instance due to not finding this AMI in the eu-west-1 region